app/src/api/controllers/editHistoryController.ts

```diff
@@ -1,17 +1,32 @@
 import express from 'express';
 
+import { ApiParamError, ApiUserError } from '../errors/api';
+import { ArgumentError } from '../errors/general';
+import { checkRegexParam, parsePositiveIntParam, processParam } from '../parameters';
 import asyncController from "../routes/asyncController";
 import * as editHistoryService from '../services/editHistory';
 
 const getGlobalEditHistory = asyncController(async (req: express.Request, res: express.Response) => {
+
+    const revisionIdRegex = /^[1-9]\d*$/;
+    const afterId: string = processParam(req.query, 'after_id', x => checkRegexParam(x, revisionIdRegex));
+    const beforeId: string = processParam(req.query, 'before_id', x => checkRegexParam(x, revisionIdRegex));
+    const count: number = processParam(req.query, 'count', parsePositiveIntParam);
+
+    if(afterId != undefined && beforeId != undefined) {
+        throw new ApiUserError('Cannot specify both after_id and before_id parameters');
+    }
+
     try {
-        const result = await editHistoryService.getGlobalEditHistory();
-        res.send({
-            history: result
-        });
+        const result = await editHistoryService.getGlobalEditHistory(beforeId, afterId, count);
+        res.send(result);
     } catch(error) {
-        console.error(error);
-        res.send({ error: 'Database error' });
+        if(error instanceof ArgumentError && error.argumentName === 'count') {
+            const apiErr = new ApiParamError(error.message, 'count');
+            throw apiErr;
+        }
```

Comment:
Map between `ArgumentError` which doesn't know anything about the service being called inside an API controller, to an API-focused `ApiParamError`. This is useful for determining a situation where the `count` parameter is correctly parsed as an integer, but it doesn't pass some constraint checks defined inside the business-logic-oriented edit history service. The mapping to an `ApiParamError` is necessary for the global error handler to send an HTTP response with the correct (user-error) code (400) rather than generic server error code (500)